File: 515-find-largest-value-in-each-tree-row/find-largest-value-in-each-tree-row.py
class Solution:
    def largestValues(self, root: Optional[TreeNode]) -> List[int]:
        if not root:
            return []

        queue, res = [root], []
        while queue:
            level_size = len(queue)
            max_val = float('-inf')
            for _ in range(level_size):
                node = queue.pop(0)
                max_val = max(max_val, node.val)
                if node.left:
                    queue.append(node.left)
                if node.right:
                    queue.append(node.right)
            res.append(max_val)
        return res

File: 515-find-largest-value-in-each-tree-row/find-largest-value-in-each-tree-row.go
func largestValues(root *TreeNode) []int {
    if root == nil {
        return []int{}
    }

    queue := []*TreeNode{root}
    res := []int{}
    for len(queue) > 0 {
        levelSize := len(queue)
        maxVal := queue[0].Val // Initialize maxVal to the value of the first node in the queue
        for i := 0; i < levelSize; i++ {
            node := queue[0]
            queue = queue[1:]
            if node.Val > maxVal {
                maxVal = node.Val
            }
            if node.Left != nil {
                queue = append(queue, node.Left)
            }
            if node.Right != nil {
                queue = append(queue, node.Right)
            }
        }
        res = append(res, maxVal)
    }
    return res
}